src/main/java/com/puppycrawl/tools/checkstyle/filters/PatchXpathFilterElement.java

```diff
@@ -144,7 +150,13 @@ private boolean isMatchingByContextStrategy(TreeWalkerAuditEvent event) {
             result = true;
         }
         else if (SUPPORT_CONTEXT_STRATEGY_CHECKS.contains(checkShortName)) {
-            final DetailAST eventAst = getEventAst(event);
+            final DetailAST eventAst;
+            if (PARAMETER_NUMBER_CHECK.equals(checkShortName)) {
```

Comment:
such hardcode is too heavy .... please be aware that where are numerous thirdparty Checks in whole word. We host only 170 of what we call standard checks that we own. Filter should work with Checks that we do not know.
for POC phase such hard code might be ok, but at the end we should not have hard code. All that should be configured by user.

For this particalar case. we should make filter smart enough to understand that some AST nodes are abstract and do not have anything related in test file. METHOD_DEF is abstract node, so all violation on it or on its children is the same 

lets make method like `isParentOfAbstractToken` and if it return true - use both nodes in match.

Reply:
>It is better to consider that Filrter should not be aware of Checks at all, all Checks names should be provided by user >in properties. We might have properties with default values that have names of Checks we know(stnadard), but user >should be always add his personal Check to list to make it work in same way.

yes, this is a problem about how this filter can work with user personal checks.

>lets make method like isParentOfAbstractToken and if it return true - use both nodes in match.

I am not clear how to know whether a node is a abstract node, please give me more information.

Reply:
@HuGanghui , please make new properties in Filter as Ia suggested to move name of Checks to config (not a hardcode in Filter)

Reply:
done

Reply:
I do not see `checkNameForContextStrategyByTokenOrParent` or anything similar in Filter.

PatchXpathFilterElement - is internal class that is not visible to user it is not a filter.


filter class SuppressionPatchXpathFilter.

config should be:
```
        <module name="com.puppycrawl.tools.checkstyle.filters.SuppressionPatchXpathFilter">
            <property name="file" value="${tp}/defaultContext.patch" />
            <property name="strategy" value="patchedline" />
            <property name="checkNameForContextStrategyByTokenOrParent" value="ParameterNumber" />
        </module>
```

NO hardcode of Check names at all in any class that is related to Filter.

Reply:
@romani I add a config here `abstractNodeChecks` https://github.com/checkstyle/patch-filters/pull/130#discussion_r463660165, I will change its name to `checkNameForContextStrategyByTokenOrParent`

Reply:
@romani done